data/contract_docs/logs.md

```diff
@@ -0,0 +1,97 @@
+    - First matching string field with name `attributes`
+    - or first matching string field with name `labels`
+
+### Additional fields
+
+- **NanoSecond Time field** - _optional_
+
+  - When the log line have sub-milli second precisions, regular time field not suitable to represent them.
+  - Field type must be non-nullable string and field name must be `tsNs`
+  - when this field detected, clients are suggested to use this field and ignore the main time field
+  - Field values can't be empty and must have only positive numbers. (no decimal places or floating numbers)
+
+If any other fields (remainder fields) found, they will be treated as items of the attributes field.
+
+## Example
+
+Following is an example of a logs frame in go
+
+```go
+data.NewFrame("logs",
+    data.NewField("timestamp", nil, []time.Time{time.UnixMilli(1645030244810), time.UnixMilli(1645030247027), time.UnixMilli(1645030247027)}),
+    data.NewField("body", nil, []string{"message one", "message two", "message three"}),
+    data.NewField("severity", nil, []string{"critical", "error", "warning"}),
+    data.NewField("id", nil, []string{"xxx-001", "xyz-002", "111-003"}),
+    data.NewField("attributes", nil, []json.RawMessage{[]byte(`{}`), []byte(`{"hello":"world"}`), []byte(`{"hello":"world", "foo": 123.45, "bar" :["yellow","red"], "baz" : { "name": "alice" }}`)}),
+    data.NewField("tsNs", nil, []string{"1645030244810757120", "1645030247027735040", "1645030247027745040"}),
+)
+```
+
+the same can be represented as
+
+| Name: timestamp <br/> Type: []time.Time | Name: body <br/> Type: []string | Name: severity <br/> Type: []string | Name: id <br/> Type: []string | Name: attributes <br/> Type: []json.RawMessage                                         | Name: tsNs <br/> Type: []string |
+| --------------------------------------- | ------------------------------- | ----------------------------------- | ----------------------------- | -------------------------------------------------------------------------------------- | ------------------------------- |
+| 2022-02-16 16:50:44.810 +0000 GMT       | message one                     | critical                            | xxx-001                       | {}                                                                                     | 1645030244810757120             |
+| 2022-02-16 16:50:47.027 +0000 GMT       | message two                     | error                               | xyz-002                       | {"hello":"world"}                                                                      | 1645030247027735040             |
+| 2022-02-16 16:50:47.027 +0000 GMT       | message three                   | warning                             | 111-003                       | {"hello":"world", "foo": 123.45, "bar" :["yellow","red"], "baz" : { "name": "alice" }} | 1645030247027745040             |
+
+## Meta data requirements
+
+- Contract doesn't require any specific meta data.
```

Comment:
Should we include a FrameType this go: "LogLines" / js: "log-lines" ? Seems so to keep it consistent.